### PR TITLE
New api helper function waitForConfirmations() - Subtask #678 - Close #900

### LINKED ISSUE #900: New api helper to wait for transactions to be confirmed
#888 is does not solved completely our problem when running several tests in parallel.

The workaround now proposed is to have a function  which has an array of transactions to query the API until they are confirmed. We set a timeout to wait to avoid never-end problems.

Parent #678

### PR DESCRIPTION
**Pull request by diego-G**
It introduces the new function and utilises it in every functional test but blocks, dapps and multisig which are being reviewed on #887, #846 & #902.

Subtask #678 
Close #900

### CODE REVIEW

File: test/common/apiHelpers.js [hunk 1]
```diff
@@ -167,6 +169,46 @@ function getBlocksToWaitPromise () {
 		});
 }
 
+function waitForConfirmations (transactions, limitHeight = 10) {
+
+	function checkConfirmations (transactions) {
+		return node.Promise.map(transactions, function (transaction) {
+			return getTransactionPromise(transaction);
+		})
+			.then(function (res) {
```
Comment (MaciejBaj) [inline test/common/apiHelpers.js]: indent looks weird

File: test/common/apiHelpers.js [hunk 2]
```diff
@@ -167,6 +169,46 @@ function getBlocksToWaitPromise () {
 		});
 }
 
+function waitForConfirmations (transactions, limitHeight = 10) {
```
Comment (MaciejBaj) [inline test/common/apiHelpers.js]: default args is es6 feature

File: test/common/apiHelpers.js [hunk 3]
```diff
@@ -167,6 +169,46 @@ function getBlocksToWaitPromise () {
...
+		return node.Promise.map(transactions, function (transaction) {
+			return getTransactionPromise(transaction);
+		})
+			.then(function (res) {
+				return node.Promise.each(res, function (result) {
+					node.expect(result).to.have.property('success').to.be.ok;
+					node.expect(result).to.have.property('transaction').not.empty;
+				})
+					.then(true)
+					.catch(function (err) {
```
Comment (MaciejBaj) [inline test/common/apiHelpers.js]: There is no reason for catching the error if you are just throwing it

File: test/common/apiHelpers.js [hunk 4]
```diff
@@ -167,6 +169,46 @@ function getBlocksToWaitPromise () {
...
+						throw err;
+					});	
+			})
+			.catch(function (err) {
```
Comment (MaciejBaj) [inline test/common/apiHelpers.js]: There is no reason for catching the error if you are just throwing it

File: test/functional/http/get/delegates.js [hunk 5]
```diff
@@ -399,27 +399,42 @@ describe('GET /api/delegates', function () {
...
 					node.expect(res).to.have.property('success').to.be.ok;
 					node.expect(res).to.have.property('transactionId').that.is.not.empty;
-					return getBlocksToWaitPromise().then(waitForBlocksPromise);
+					transactionsToWaitFor = [];
+					transactionsToWaitFor.push(res.transactionId);
+				})
+				.then(function (res) {
+					return waitForConfirmations(transactionsToWaitFor);
+				})
+				.catch(function (err) {
```
Comment (SargeKhan) [inline test/functional/http/get/delegates.js]: Dont need this

File: test/functional/http/get/transactions.js [hunk 6]
```diff
@@ -36,15 +35,21 @@ describe('GET /api/transactions', function () {
...
+				results.forEach(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').that.is.not.empty;
+					transactionList.push(res.transactionId);
+				});
+			})
+			.then(function (res) {
+				return waitForConfirmations(transactionList);
+			})
+			.catch(function (err) {
```
Comment (SargeKhan) [inline test/functional/http/get/transactions.js]: same

File: test/functional/shared.js [hunk 7]
```diff
@@ -35,7 +34,14 @@ function confirmationPhase (goodTransactions, badTransactions, pendingMultisigna
 	describe('after transactions get confirmed', function () {
 
 		before(function () {
-			return getBlocksToWaitPromise().then(waitForBlocksPromise);
+			var transactionToWaitFor = goodTransactions.map(function (transaction) {
+				return [transaction.id];
+			});
+			return waitForConfirmations(transactionToWaitFor)
+				.catch(function (err) {
```
Comment (SargeKhan) [inline test/functional/shared.js]: same

File: test/common/apiHelpers.js [hunk 8]
```diff
@@ -167,6 +169,46 @@ function getBlocksToWaitPromise () {
...
 
+function waitForConfirmations (transactions, limitHeight = 10) {
+
+	function checkConfirmations (transactions) {
+		return node.Promise.map(transactions, function (transaction) {
+			return getTransactionPromise(transaction);
+		})
+			.then(function (res) {
+				return node.Promise.each(res, function (result) {
+					node.expect(result).to.have.property('success').to.be.ok;
```
Comment (MaciejBaj) [inline test/common/apiHelpers.js]: Is `expect` throwing an error? I think the better idea is to throw it directly: `if (result.success === false) throw 'Not confirmed'`

File: test/common/apiHelpers.js [hunk 9]
```diff
@@ -169,25 +169,20 @@ function getBlocksToWaitPromise () {
...
-					.then(true)
-					.catch(function (err) {
-						throw err;
-					});	
-			})
-			.catch(function (err) {
-				throw err;
-			});	
+					if (result.success === false) {
+						throw Error(result.error);
```
Comment (SargeKhan) [inline test/common/apiHelpers.js]: Why not something like: `node.expect(result).to.have.property('success').to.be.ok;`?

Comment (diego-G) [reply test/common/apiHelpers.js]: @MaciejBaj suggested to use code instead of messing around with Mocha code. I am ok with both solutions.

**Pull request merged**